src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +964,767 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
             return dbResultSetRow;
         }
 
+        /// <summary>
+        /// Performs the given GraphQL create mutation operation.
+        /// </summary>
+        /// <param name="entityName">Name of the top level entity</param>
+        /// <param name="parameters">Mutation parameter arguments</param>
+        /// <param name="sqlMetadataProvider">SqlMetadaprovider</param>
+        /// <param name="context">Hotchocolate's context for the graphQL request.</param>
+        /// <param name="multipleInputType">Boolean indicating whether the create operation is for multiple items.</param>
+        /// <returns>Primary keys of the created records (in the top level entity).</returns>
+        /// <exception cref="DataApiBuilderException"></exception>
+        private List<IDictionary<string, object?>> PerformMultipleCreateOperation(
+                string entityName,
+                IMiddlewareContext context,
+                IDictionary<string, object?> parameters,
+                ISqlMetadataProvider sqlMetadataProvider,
+                bool multipleInputType = false)
+        {
+            string fieldName = multipleInputType ? MULTIPLE_INPUT_ARGUEMENT_NAME : SINGLE_INPUT_ARGUEMENT_NAME;
+
+            // Parse the hotchocolate input parameters into .net object types
+            object? parsedInputParams = GQLMultipleCreateArgumentToDictParams(context, fieldName, parameters);
+
+            if (parsedInputParams is null)
+            {
+                throw new DataApiBuilderException(message: "The input for multiple create mutation operation cannot be null",
+                                                  statusCode: HttpStatusCode.BadRequest,
+                                                  subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+            }
+
+            // List of Primary keys of the created records in the top level entity.
+            // Each dictionary in the list corresponds to the PKs of a single record.
+            // For point multiple create operation, only one entry will be present.
+            List<IDictionary<string, object?>> primaryKeysOfCreatedItemsInTopLevelEntity = new();
+
+            if (!parameters.TryGetValue(fieldName, out object? param) || param is null)
```

Comment:
need more contextual name than `param`. `param` is more symbolic than just any param right? it's the field `item` or `items` and all input metadata associated with that field

Reply:
Updated 

`parameters` --> `mutationInputParamsFromGQLContext`
`param` --> `unparsedInputFieldsForRootField`